steps/generate_dataset_from_htme.py

```diff
@@ -38,13 +38,14 @@ def get_list_keys_for_prefix():
     response = s3_client.list_objects_v2(Bucket=s3_htme_bucket, Prefix=s3_prefix)
     for obj in response.get("Contents", []):
         keys.append(obj["Key"])
+    keys.remove(s3_prefix)
     return keys
 
 
 def group_keys_by_collection(keys):
-    file_key_dict = {key.split("/")[-1]: key for key in keys}
+    file_key_dict = {key.split("/")[5]: key for key in keys}
```

Comment:
I don't like this hardcoded number as it's very fragile. What are we trying to do here? Is it not just trying to get the last portion of the s3 prefix, i.e. the filename? In which case, why doesn't the `-1` index work?

Reply:
-1 also works, have change the code to reflect the same